### PR TITLE
[Compose] - 2565 - Add channel display name factory

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2565

### 🎯 Goal

Add `Channel` display name factory customization option to the app.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF

giphy (2)

### CODE REVIEW

**Review by filbabic — APPROVED** [review]
Looks good to me! :]

**Review by filbabic — CHANGES_REQUESTED** [review]
One thing I did notice is - we're missin Kdocs for the new provider and parameter in ChatTheme as well as docusaurus.

Can you please update those too! (sorry if I missed them initially)

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelNameFormatter.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * - If the channel is distinct, then a comma-separated list of member names is returned
+ * - Otherwise, the placeholder text defined in [R.string.stream_compose_untitled_channel] is returned
+ *
+ * @param context The context to load string resources.
+ * @param maxMembers The maximum number of members used to generate a name for a distinct channel.
+ */
+private class DefaultChannelNameFormatter(
+    private val context: Context,
+    private val maxMembers: Int,
+) : ChannelNameFormatter {
```
Comment (skydoves) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelNameFormatter.kt:53]: I'm not sure but how about adding the `@Composable` annotation to the `format` method and removing the `context `argument? We may get the context from the `LocalContext.current` instead.

```kotlin
@Composable
override fun format(channel: Channel): String {
    val context: Context = LocalContext.current
    return channel.name.takeIf { it.isNotEmpty() }
        ?: channel.getUsersExcludingCurrent()
            .joinToString(limit = maxMembers) { it.name }
            .takeIf { it.isNotEmpty() }
        ?: context.getString(R.string.stream_compose_untitled_channel)
}
```

**Review by skydoves — COMMENTED** [review]
Totally looks good to me! I left one suggestion. :)

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelNameFormatter.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelNameFormatter.kt:53]: That's actually a very good question. I actually don't have a strong opinion about that.

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelNameFormatter.kt:53]: @skydoves @bychkovdmitry Really nice discussion.

Generally, adding @Composable to functions that return values is not that common and should serve more less only for functions that really do depend on the scope and context of the Compose tree.

E.g. if you're observing state that changes and the function needs to recompose. Or if you're, based on the compose state, lauching extra lifecycle things that affect the Compose tree in some way, like system back click handlers updating the navigation or the active screen.

In this case, although we depend on the Context, nothing mutates within the Compose tree to affect our state or result, as this result will always be the same, regardless of the context object (it's just used to fetch the resource).

So in that sense, I prefer it this way and I think it's good, maybe we could split it up into a smaller function that might be more descriptive, but it's also fine this way!

**Review by filbabic — APPROVED** [review]
Looks good!

**Review by skydoves — APPROVED** [review]

**Pull request merged**